src/main/java/tech/autodirect/api/database/TableOffers.java

```diff
@@ -0,0 +1,219 @@
+        Statement stmt = this.db_conn.createStatement();
+        stmt.executeUpdate(
+        "CREATE TABLE IF NOT EXISTS " + this.table_name + " (" +
+                "offer_id       serial      NOT NULL PRIMARY KEY, " +
+                "car_id         integer     NOT NULL, " +
+                "loan_amount    decimal(12) NOT NULL, " +
+                "capital_sum    decimal(12) NOT NULL, " +
+                "interest_sum   decimal(12) NOT NULL, " +
+                "total_sum      decimal(12) NOT NULL, " +
+                "interest_rate  real        NOT NULL, " +
+                "term_mo        real        NOT NULL, " +
+                "installments   jsonb       NOT NULL, " +
+                "claimed        boolean     NOT NULL" +
+            ");"
+        );
+        stmt.close();
+    }
+
+    public int addOffer(
+        int car_id,
+        BigDecimal loan_amount,
+        BigDecimal capital_sum,
+        BigDecimal interest_sum,
+        BigDecimal total_sum,
+        double interest_rate,
+        double term_mo,
+        String installments,
+        boolean claimed
+    ) throws SQLException {
+        // construct a prepared SQL statement inserting the specified values
+        PreparedStatement stmt = this.db_conn.prepareStatement(
+        "INSERT INTO " + this.table_name + " (" +
+                String.join(", ", table_columns) +
+            ")" +
+                "VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?);"
+        );
+        stmt.setInt(1, car_id);
+        stmt.setBigDecimal(2, loan_amount);
+        stmt.setBigDecimal(3, capital_sum);
+        stmt.setBigDecimal(4, interest_sum);
+        stmt.setBigDecimal(5, total_sum);
+        stmt.setDouble(6, interest_rate);
+        stmt.setDouble(7, term_mo);
+        stmt.setString(8, installments);
+        stmt.setBoolean(9, claimed);
+
+        // execute and close the above SQL statement
+        stmt.executeUpdate();
+        stmt.close();
+
+        // retrieve the offer ID
+        Statement stmt_lastval = this.db_conn.createStatement();
+        ResultSet rs = stmt_lastval.executeQuery("SELECT LASTVAL();");
+        rs.next();
+        int offer_id = rs.getInt("lastval");
+        stmt.close();
+
+        return offer_id;
+    }
+
+    public void removeOfferByOfferId(int offer_id) throws SQLException {
+        // construct a prepared SQL statement deleting the specified offer
+        PreparedStatement stmt = this.db_conn.prepareStatement(
+        "DELETE FROM " + this.table_name + " WHERE offer_id = ?;"
+        );
+        stmt.setInt(1, offer_id);
+
+        // execute and close the above SQL statement
+        stmt.executeUpdate();
+        stmt.close();
+    }
+
+    public void removeAllOffers() throws SQLException {
+        Statement stmt = this.db_conn.createStatement();
+        stmt.executeUpdate("DELETE FROM " + this.table_name + ";");
+        stmt.close();
+    }
+
+    private static HashMap offerResultToHashMap(ResultSet rs) throws SQLException {
```

Comment:
Raw use of HashMap, consider replacing with HashMap<String, Object> as its better for documentation purposes